### PR TITLE
#165210376 Filter meeting rooms by wings and floors

### PR DESCRIPTION
**Pull request by TeamoreA**
### What does this PR do?
Adds a `roomLabels` argument to `allRooms` query that filters rooms by wings and floors
### Description of the task to be completed?
When a user provides the argument `roomLabels` to the `allRooms` query the query returns all rooms that have that label. One can also provide comma separated labels to perform a deeper filter and only rooms that contain the provided labels will be returned
### How should this be manually tested?
 - Clone the branch and follow the following steps to [setup](https://github.com/andela/mrm_api).
 - Checkout by running `gco ft-filter-meeting-rooms-by-wings-and-floors-165210376`
 - Perform the mutation below to query rooms with the specified labels.
```
{
  allRooms(roomLabels:"Block A, Wing B"){
    rooms{
      id
      name
      roomLabels
    }
  }
}
```
### Any background context you want to provide?
You must create rooms with the room labels you want to query with, otherwise, no rooms will be returned.
### What are the relevant pivotal tracker stories?
[#165210376](https://www.pivotaltracker.com/story/show/165210376)
### Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have linted my code prior to submission
- [x] My changes generate no new warnings
- [x] Existing unit tests pass locally with my changes
- [x] Implementation works according to expectations
### Screenshots
- #### Query for all rooms with `roomLabels` filter
image

### CODE REVIEW

**Comment by TeamoreA** [comment]
> Good work @TeamoreA I have a question though
> What if a room has only one of the labels in the query, shouldn't the query return this room too?
> At the moment I found it returns the room with exact labels in the query

Yeah, it returns all rooms with the specified room label if you only provide one label. Kindly test it out again with several rooms.

**Comment by rajeman** [comment]
@TeamoreA I found out the tags are case sensitive. Shouldn't they be case insensitive? i.e. if I pass `block A` instead of `Block A`, I feel it should give the same result.

**Comment by TeamoreA** [comment]
> @TeamoreA I found out the tags are case sensitive. Shouldn't they be case insensitive? i.e. if I pass `block A` instead of `Block A`, I feel it should give the same result.
Since the filter labels will be supplied from a frontend dropdown that will still be populated from the backend, case insensitivity wouldn't be important for this case. @rajeman

**Comment by d-kahara** [comment]
> > @TeamoreA I found out the tags are case sensitive. Shouldn't they be case insensitive? i.e. if I pass `block A` instead of `Block A`, I feel it should give the same result.
> > Since the filter labels will be supplied from a frontend dropdown that will still be populated from the backend, case insensitivity wouldn't be important for this case. @rajeman

Considering that we're building an API which could possibly be integrated with other APIs, not only front-end frameworks, I think @rajeman 's observation is valid.

File: tests/test_rooms/test_room_filter.py [hunk 1]
```diff
@@ -36,3 +38,10 @@ def test_filter_room_by_location_capacity(self):
             filter_rooms_by_location_capacity,
             filter_rooms_by_location_capacity_response
         )
+
+    def test_filter_room_by_wings_and_floors(self):
```
Comment (Bonifase) [inline tests/test_rooms/test_room_filter.py]: Write tests for all the edge cases, i.e when there is no match and no results, when the filter parameter is not valid. Also, take care of the case sensitivity

File: api/room/schema_query.py [hunk 2]
```diff
@@ -115,6 +115,7 @@ class Query(graphene.ObjectType):
     """
     all_rooms = graphene.Field(
         PaginatedRooms,
+        room_labels=graphene.String(),
```
Comment (jesseinit) [inline api/room/schema_query.py]: Good work on the PR @TeamoreA,
Kindly add room label to the description on line `126`

Comment (TeamoreA) [reply api/room/schema_query.py]: This has been added @jesseinit

File: tests/test_rooms/test_room_filter.py (hunk 1 above)
Comment (TeamoreA) [reply tests/test_rooms/test_room_filter.py]: This has been worked on @Bonifase

**Comment by rajeman** [comment]
Working as expected. Well done
